finality-aleph/src/testing/data_store.rs

```diff
@@ -442,6 +448,38 @@ async fn sends_block_request_on_missing_block() {
     .await;
 }
 
+#[tokio::test]
+async fn sends_justification_request_when_not_finalized() {
```

Comment:
I'm just thinking out loud: we have here multiple scenarios testing different ordering of events; had we checked this particular configuration before, we would be aware of the problem before outage

and hence my random thought: is it possible/feasible to automatically and deftly test all event orderings? or at least increase coverage somehow?